party_menu/party_menu.lua

```diff
@@ -0,0 +1,469 @@
+local status_effects = require('status_effects')
+local packets = require('packets')
+local chat = require('chat')
+local command = require('command')
+local settings = require('settings')
+local target = require('target')
+local party = require('party')
+local math = require('math')
+local os = require('os')
+local enumerable = require('enumerable')
+
+
+local debuffs = require('debuff_priority_list')
+local buffs = require('buff_list')
+local all_statuses = require('all_status_list')
+local status_list = {debuffs, buffs, all_statuses}
+
+local hp_modes = {'missinghp', 'hp', 'hp/hp_max', 'hide'}
+local mp_modes = {'mp', 'mp/mp_max', 'hide'}
+local status_modes = {'debuffs', 'buffs', 'all statuses', 'no statuses'}
+
+local zoning = false
+local zone_out_time = os.clock()
+local zone_in_delay = 10
+
+local defaults = {
+    names_visible = true,
+    hp_mode_index = 1,
+    mp_mode_index = 3,
+    status_mode_index = 1,
+    tp_visible = false,
+    dynamic_coloring = true,
+    text_size = 20,
+    text_typeface = 'Consolas',
+    global_window_state = {
+        style = 'chromeless',
+        color = ui.color.black,
+        closable = true,
+        x = 20,
+        y = 100,
+        width = 500,
+    }
+}
+
+local global_window_state = {}
+
+settings.settings_change:register(function()
+    for key, value in pairs(defaults.global_window_state) do
+        global_window_state[key] = value
+    end
+end)
+
+local options = settings.load(defaults)
+
+local global_window_closed = false
+local outer_padding = 2
+local left_inner_padding = 2
+local upper_inner_padding = 2
+local lower_inner_padding = 5
+local row_window_height = options.text_size + upper_inner_padding + lower_inner_padding
+global_window_state.height = party.alliance.party_1_count * (row_window_height + outer_padding) + outer_padding
+
+local get_name = function(pos)
+    if options.names_visible then
+        return string.format('%5.5s ', party[pos].name)
+    else
+        return ''
+    end
+end
+
+local get_hp = function(pos)
+    if hp_modes[options.hp_mode_index] == 'hide' then
+        return ''
+    elseif party[pos].hp == 0 and hp_modes[options.hp_mode_index] == 'missinghp' then
+        return 'mHP:' .. string.format('%-5.5s ', 'DEAD')
+    elseif party[pos].hp == 0 and hp_modes[options.hp_mode_index] == 'hp/hp_max' then
+        return 'HP:' .. string.format('%-9.9s ', 'DEAD')
+    elseif party[pos].hp == 0 then
+        return 'HP:' .. string.format('%4.4s ', 'DEAD')
+    elseif hp_modes[options.hp_mode_index] == 'missinghp' then
+        return 'mHP:-' .. string.format('%-4.4s ', math.floor(party[pos].hp * 100 / party[pos].hp_percent) - party[pos].hp)
+    elseif hp_modes[options.hp_mode_index] == 'hp/hp_max' then
+        return 'HP:' .. string.format('%4.4s', party[pos].hp) .. '/' .. string.format('%-4.4s ', math.floor(party[pos].hp * 100 / party[pos].hp_percent))
+    else
+        return 'HP:' .. string.format('%4.4s ', party[pos].hp)
```

Comment:
This is going to bite you if a player has more than 9999 hp, which I think is possible in Abyssea if you eat food.